pkg/cephfs/util.go

```diff
@@ -30,61 +30,41 @@ import (
 
 	"github.com/ceph/ceph-csi/pkg/util"
 	"github.com/container-storage-interface/spec/lib/go/csi"
+	"k8s.io/kubernetes/pkg/util/keymutex"
 	"k8s.io/kubernetes/pkg/util/mount"
 )
 
 type volumeID string
 
-func makeVolumeID(volName string) volumeID {
-	return volumeID("csi-cephfs-" + volName)
+func mustUnlock(m keymutex.KeyMutex, key string) {
+	if err := m.UnlockKey(key); err != nil {
```

Comment:
Unlock  always returns nil any idea why?
https://github.com/ceph/ceph-csi/blob/csi-v1.0/vendor/k8s.io/kubernetes/pkg/util/keymutex/hashed.go#L57

```
// Releases the lock associated with the specified ID.
func (km *hashedKeyMutex) UnlockKey(id string) error {
	klog.V(5).Infof("hashedKeyMutex.UnlockKey(...) called for id %q\r\n", id)
	km.mutexes[km.hash(id)%len(km.mutexes)].Unlock()
	klog.V(5).Infof("hashedKeyMutex.UnlockKey(...) for id %q completed.\r\n", id)
	return nil
}
```

Reply:
Just a guess but maybe this is a deliberate decision to `panic()` on a non-existent mutex. `mutex.Lock/Unlock` doesn't return an error anyway, but rather aborts internally: https://golang.org/src/sync/mutex.go